src/TNetwork.cpp

```diff
@@ -322,12 +326,30 @@ std::shared_ptr<TClient> TNetwork::Authentication(TConnection&& RawConnection) {
     }
 
     if (mServer.ClientCount() < size_t(Application::Settings.MaxPlayers)) {
+
+        if(!Application::Settings.Password.empty()) { // ask password
```

Comment:
Password check **needs** to happen before `onPlayerAuth`, because `onPlayerAuth is supposed to be called only once everything except player count is checked - so this needs to be changed (moved to 285). We want the client to cause as little state change as possible.

Reply:
I would beg to disagree, you authenticate then you know who you are dealing with, it adds extra security and you know who failed to connect by name

Reply:
You are doing the password check after calling onPlayerAuth, which will break existing plugins, please move it to the desired location.

onPlayerAuth assumes that it is the last step in authentication, and that if it succeeds, the player will join if the player limit allows it. 

The player should not be made known to lua before passing the password check, as this already introduces the player to the system - we want to block the player as early as possible.

Reply:
Make sure you read again which line in the code im referencing - im saying to move it after authentication, before the call to onPlayerAuth

Reply:
So since you were not clear enough you mean check before lua auth call but after the actual auth check against the backend?

Reply:
ok then I will blindly put it where you want then